pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -359,17 +359,38 @@ func TestPriorityQueue_Delete(t *testing.T) {
 }
 
 func TestPriorityQueue_MoveAllToActiveOrBackoffQueue(t *testing.T) {
-	q := NewPriorityQueue(newDefaultQueueSort())
+	c := clock.NewFakeClock(time.Now())
+	q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c))
 	q.Add(&medPriorityPod)
+	q.MoveAllToActiveOrBackoffQueue("test")
+	// Make sure the move request is happened first.
+	c.Step(time.Second)
```

Comment:
The next two lines for `AddUnschedulableIfNotPresent`, they depend on the `PriorityQueue.moveRequestCycle` and `PriorityQueue.schedulingCycle` to decide going to the backoff queue or unschedule queue. So this should be useless, it is only useful for `MoveAllToActiveOrBackoffQueue`, it needs the clock to determine the pod is backoff or not.

Reply:
> So this should be useless, it is only useful for MoveAllToActiveOrBackoffQueue, it needs the clock to determine the pod is backoff or not.

sorry, I am quite follow you here.  In this testcase, all that `MoveAllToActiveOrBackoffQueue` did is set `p.moveRequestCycle`,  I cannot see `MoveAllToActiveOrBackoffQueue` needs the clock anywhere.

Reply:
> > So this should be useless, it is only useful for MoveAllToActiveOrBackoffQueue, it needs the clock to determine the pod is backoff or not.
> 
> sorry, I am quite follow you here. In this testcase, all that `MoveAllToActiveOrBackoffQueue` did is set `p.moveRequestCycle`, I cannot see `MoveAllToActiveOrBackoffQueue` needs the clock anywhere.

Sorry, I didn't describe clearly.  :)

I mean `AddUnschedulableIfNotPresent` needn't the clock, so you needn't step the clock here

We only need to step the clock when you test the `MoveAllToActiveOrBackoffQueue `, like line 385. It needs the clock to determine pod going to which queue:

`https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/queue/scheduling_queue.go#L514`

`https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/queue/scheduling_queue.go#L281`

Reply:
Done